[17.0][OU-ADD] pos_restaurant: Migration to 17.0

carlos-lopez-tecnativa opened the pull request:
TT57261
@Tecnativa @pedrobaeza @MiquelRForgeFlow could you please review this?

carlos-lopez-tecnativa committed:
[OU-ADD] pos_restaurant: Migration to 17.0

pedrobaeza commented:
/ocabot migration pos_restaurant

pedrobaeza commented on openupgrade_scripts/scripts/pos_restaurant/17.0.1.0/upgrade_analysis_work.txt:
```diff
@@ -0,0 +1,46 @@
+# NOTHING TO DO: new feature used to install the pos_restaurant_appointment module in Enterprise
+
+pos_restaurant / pos.order                / multiprint_resume (char)      : DEL
+pos_restaurant / pos.order.line           / mp_skip (boolean)             : DEL
```
The equivalent field is now `skip_line`, so a renaming is needed, take into account that now the field is in `point_of_sale`:

https://github.com/odoo/odoo/commit/73aae73f1ba4ad9471d0cded3d8f02cac8cc3669#diff-29cbaebb5b63b539ab173d9340b2aec87b9ad63cd322eec2347879c5412bd50bR1286

carlos-lopez-tecnativa replied:
I updated the `upgrade_analysis_work` to indicate that this is handled in the `point_of_sale` module migration scripts: https://github.com/OCA/OpenUpgrade/blob/03bd8c9c4be30e276162ebf861cd6a5f4527df41/openupgrade_scripts/scripts/point_of_sale/17.0.1.0.1/pre-migration.py#L13

pedrobaeza commented on openupgrade_scripts/scripts/pos_restaurant/17.0.1.0/upgrade_analysis_work.txt:
```diff
@@ -0,0 +1,46 @@
+pos_restaurant / pos.config               / module_pos_restaurant_appointment (boolean): NEW
+# NOTHING TO DO: new feature used to install the pos_restaurant_appointment module in Enterprise
+
+pos_restaurant / pos.order                / multiprint_resume (char)      : DEL
```
This field has been renamed into:

https://github.com/odoo/odoo/commit/73aae73f1ba4ad9471d0cded3d8f02cac8cc3669#diff-29cbaebb5b63b539ab173d9340b2aec87b9ad63cd322eec2347879c5412bd50bR241

carlos-lopez-tecnativa replied:
I made this change in this PR but in a separate commit. Please let me know if it’s OK or if I need to create a new PR.

pedrobaeza commented on openupgrade_scripts/scripts/pos_restaurant/17.0.1.0/upgrade_analysis_work.txt:
```diff
@@ -0,0 +1,46 @@
+
+pos_restaurant / pos.config               / is_order_printer (boolean)    : module is now 'point_of_sale' ('pos_restaurant')
+pos_restaurant / pos.config               / printer_ids (many2many)       : module is now 'point_of_sale' ('pos_restaurant')
+pos_restaurant / pos.config               / printer_ids (many2many)       : relation is now 'pos.printer' ('restaurant.printer') [nothing to do]
```
Uhm, weird that this is detected as a change of the model, isn't it?

carlos-lopez-tecnativa replied:
I don’t understand what you mean. In the last line, the analysis indicates this: `relation is now 'pos.printer' ('restaurant.printer') [nothing to do] `

carlos-lopez-tecnativa committed:
[OU-IMP] point_of_sale: correct implementation of the last_order_preparation_change field

According to the analysis, this initially looked like a new feature, but upon closer examination it is actually a field rename from multiprint_resume in the pos_restaurant model. This commit renames the field to preserve existing values.
